source/http_headers.c

```diff
@@ -96,18 +96,19 @@ PyObject *aws_py_http_headers_add_pairs(PyObject *self, PyObject *args) {
         return NULL;
     }
 
-    const Py_ssize_t count = PySequence_Fast_GET_SIZE(py_sequence);
+    const Py_ssize_t count = PySequence_Size(py_sequence);
     for (Py_ssize_t i = 0; i < count; ++i) {
         /* XYZ_GET_ITEM() calls returns borrowed references */
-        PyObject *py_pair = PySequence_Fast_GET_ITEM(py_sequence, i);
+        PyObject *py_pair = PySequence_GetItem(py_sequence, i);
 
-        if (!PyTuple_Check(py_pair) || PyTuple_GET_SIZE(py_pair) != 2) {
+        if (!PyTuple_Check(py_pair) || PyTuple_Size(py_pair) != 2) {
             PyErr_SetString(PyExc_TypeError, type_errmsg);
             goto done;
```

Comment:
if we take this branch, we leak

Reply:
Discussed offline, that PyTuple steals a reference.